### PR TITLE
update dynamic file sink

### PR DESCRIPTION
**Pull request by nevillelyh**

### CODE REVIEW

File: scio-core/src/main/scala/com/spotify/scio/values/SCollection.scala [hunk 1]
```diff
@@ -960,8 +960,8 @@ sealed trait SCollection[T] extends PCollectionWrapper[T] {
    * @group output
    */
   def saveAsTextFile(path: String,
-                     suffix: String = ".txt",
                      numShards: Int = 0,
+                     suffix: String = ".txt",
```
Comment (nevillelyh) [inline scio-core/src/main/scala/com/spotify/scio/values/SCollection.scala]: This is technically breaking but it'll be a breaking change anyway.
It's to make the API consistent with other `saveAs*` methods.

File: scio-core/src/main/scala/com/spotify/scio/io/dynamic/package.scala [hunk 2]
```diff
@@ -123,27 +97,27 @@ package object dynamic {
...
-      saveAsTextFile(destinations, fileDestination.windowedWrites, fileDestination.numShards)
-    }
-
   }
 
+
+  private def writeDynamic[A](dest: FileDestinations,
+                              suffix: String,
+                              destinationFn: A => String): FileIO.Write[String, A] = {
+    FileIO.writeDynamic[String, A]()
```
Comment (regadas) [inline scio-core/src/main/scala/com/spotify/scio/io/dynamic/package.scala]: @nevillelyh should we add a compression option?

File: scio-core/src/main/scala/com/spotify/scio/values/SCollection.scala (hunk 1 above)
Comment (regadas) [reply scio-core/src/main/scala/com/spotify/scio/values/SCollection.scala]: @nevillelyh I'm ok with that!

**Comment by nevillelyh** [comment]
Think I need to rename these 2 `saveAs*` methods since they have default args and don't play well with the ones on `SCollection` with the same name.

**Pull request merged**